lib/internal/modules/esm/hooks.js

```diff
@@ -146,7 +147,7 @@ class Hooks {
    * loader (user-land) to the worker.
    */
   async register(urlOrSpecifier, parentURL, data) {
-    const moduleLoader = require('internal/process/esm_loader').esmLoader;
+    const moduleLoader = createModuleLoader();
```

Comment:
Was this change necessary for it to work? The existing code was done this way for a reason (to reduce node's startup time by loading the dependency just in time). You'll see that done in many places within this file.